src/models/multiBarChart.js

```diff
@@ -179,11 +179,6 @@ nv.models.multiBarChart = function() {
             if (showLegend) {
                 legend.width(availableWidth - controlWidth());
 
-                if (multibar.barColor())
-                    data.forEach(function(series,i) {
-                        series.color = d3.rgb('#ccc').darker(i * 1.5).toString();
-                    });
```

Comment:
I understand and completely agree, hence why I made the changes in commits 6031909 and 01b2804. This way, the shades of gray are the default and the colouring of the legend (in gray) is retained when barColor is used.
The only difference from what you describe is that now we also have the option to override the shades of gray colouring of the legend by using color() if we so desire. Otherwise, the behaviour should match what you describe. As I said, check out the last two commits. It may be that I am not explaining things right and just looking as the code will make it more obvious. I can provide screenshots again if that helps.